### PR TITLE
Implement a new workload hints functionality

### PR DESCRIPTION
**Pull request by cynepco3hahue**
The new API should make it easier to configure nodes for specific types of workloads via the performance profile.

### How does it look?
```
spec:
  workloadHints:
    realTime: true
...
spec:
  workloadHints:
    highPowerConsumption: false
```

### How does it work?

The tuned profile has:

Have a separate set of kernel arguments for each workload hint.
Set sysctl, cpu and scheduler section parameters according to
workload hints.

### CODE REVIEW

File: pkg/apis/performanceprofile/v1/performanceprofile_types.go [hunk 1]
```diff
@@ -78,7 +81,7 @@ type CPUSet string
 // CPU defines a set of CPU related features.
 type CPU struct {
 	// Reserved defines a set of CPUs that will not be used for any container workloads initiated by kubelet.
-	Reserved *CPUSet `json:"reserved,omitempty"`
+	Reserved *CPUSet `json:"reserved"`
```
Comment (fromanirh) [inline pkg/apis/performanceprofile/v1/performanceprofile_types.go:84]: this is probably correct but should not be done in this PR. Or if it is, is not clear why and should be done on a separate commit with proper rationale

**Review by fromanirh — COMMENTED** [review]
LGTM - one pretty minor comment inside
I'll need to review again the hints implementation proper, but the logic per se seems fine

File: assets/performanceprofile/tuned/openshift-node-performance [hunk 2]
```diff
@@ -121,12 +129,24 @@ avc_cache_threshold=8192
...
 initrd_dst_img=
 initrd_add_dir=
+
 # overrides cpu-partitioning cmdline
-cmdline_cpu_part=+nohz=on rcu_nocbs=${isolated_cores} tuned.non_isolcpus=${not_isolated_cpumask} intel_pstate=disable nosoftlockup
+cmdline_cpu_part=+nohz=on rcu_nocbs=${isolated_cores} tuned.non_isolcpus=${not_isolated_cpumask} systemd.cpu_affinity=${not_isolated_cores_expanded} intel_iommu=on iommu=pt
+
 {{if .StaticIsolation}}
-cmdline_realtime=+tsc=nowatchdog intel_iommu=on iommu=pt isolcpus=domain,managed_irq,${isolated_cores} systemd.cpu_affinity=${not_isolated_cores_expanded}
+cmdline_isolation=+isolcpus=domain,managed_irq,${isolated_cores}
```
Comment (fromanirh) [inline assets/performanceprofile/tuned/openshift-node-performance:137]: the only other thing I'm not comfortable with in this PR is that in this commit we are both changing the profiles proper and the way we create them. I would be happier if, if possible, we can have 1. a commit which takes the current profiles unchanged and just changes the way we combine the existing data, using the workload hints. 2. a commit which changes the profiles per se, but NOT the way we combine the data. IOW, I'd love to split the responsabilities in this commit.
This is NOT BLOCKING, depending on what the other reviewers prefer.

File: pkg/apis/performanceprofile/v1/performanceprofile_types.go (hunk 1 above)
Comment (cynepco3hahue) [reply pkg/apis/performanceprofile/v1/performanceprofile_types.go]: done

File: pkg/performanceprofile/controller/performanceprofile/components/tuned/tuned.go [hunk 3]
```diff
@@ -166,6 +168,21 @@ func NewNodePerformance(profile *performancev2.PerformanceProfile) (*tunedv1.Tun
 		}
 	}
 
+	// TODO: I think we need to parse additional kernel arguments and in case when it intersect with workload hints
+	// kernel arguments we should or override it or drop an error
+	// set default values for workload hints
+	templateArgs[templateRealTime] = "true"
+	templateArgs[templateHighPowerConsumption] = "true"
```
[the remark this answers is not in the document]
Comment (cynepco3hahue) [reply pkg/performanceprofile/controller/performanceprofile/components/tuned/tuned.go]: done

File: assets/performanceprofile/tuned/openshift-node-performance (hunk 2 above)
Comment (cynepco3hahue) [reply assets/performanceprofile/tuned/openshift-node-performance:135]: done

**Review by fromanirh — APPROVED** [review]
/lgtm
/hold
